src/Illuminate/Database/Eloquent/Relations/BelongsTo.php

```diff
@@ -128,7 +128,14 @@ protected function getEagerModelKeys(array $models)
         // execute a "where in" statement to gather up all of those related records.
         foreach ($models as $model) {
             if (! is_null($value = $model->{$this->foreignKey})) {
-                $keys[] = $value;
+                switch ($this->related->getKeyType()) {
```

Comment:
No need to call this method within the loop, could retrieve the value before looping.

Reply:
Got it, updated the PR. Thanks.